tests/e2e/test_rosacli_upgrade.go

```diff
@@ -314,6 +309,66 @@ var _ = Describe("Cluster Upgrade testing",
 			err = WaitForUpgradeToState(upgradeService, clusterID, con.Started, 70)
 			Expect(err).To(BeNil())
 		})
+
+		It("to upgrade STS rosa cluster across Y stream - [id:55883]", labels.Critical, labels.Runtime.Upgrade, func() {
+
+			By("Check the cluster version and compare with the profile to decide if skip this case")
+			jsonData, err := clusterService.GetJSONClusterDescription(clusterID)
+			Expect(err).To(BeNil())
+			clusterVersion := jsonData.DigString("version", "raw_id")
+
+			scheduledDate := time.Now().Format("2006-01-02")
+			scheduledTime := time.Now().Add(10 * time.Minute).UTC().Format("15:04")
+
+			if profile.Version != con.YStreamPreviousVersion || !profile.ClusterConfig.STS {
+				Skip("Skip this case as the version defined in profile is not y-1 for sts cluster upgrading testing")
+			}
+
+			if profile.ClusterConfig.HCP {
+				By("Find HCP cluster upgrade version")
+				hcpUpgradingVersion, _, err := FindUpgradeVersion(versionService,
+					profile.ChannelGroup, true, clusterVersion)
+				Expect(err).To(BeNil())
+				Expect(hcpUpgradingVersion).NotTo(BeEmpty())
+
+				By("Upgrade HCP cluster")
+				output, err := upgradeService.Upgrade(
+					"-c", clusterID,
+					"--version", hcpUpgradingVersion,
+					"--schedule-date", scheduledDate,
+					"--schedule-time", scheduledTime,
+					"--control-plane",
+					"-m", "auto",
+					"-y",
+				)
+				Expect(err).To(BeNil())
+				Expect(output.String()).To(ContainSubstring("Upgrade successfully scheduled for cluster"))
+			} else {
+				By("Find STS Classic cluster upgrade version")
+				classicUpgradingVersion, _, err := FindUpgradeVersion(versionService,
+					profile.ChannelGroup, false, clusterVersion)
+				Expect(err).To(BeNil())
+				Expect(classicUpgradingVersion).NotTo(BeEmpty())
+
+				By("Upgrade STS classic cluster")
+				output, err := upgradeService.Upgrade(
+					"-c", clusterID,
+					"--version", classicUpgradingVersion,
+					"--schedule-date", scheduledDate,
+					"--schedule-time", scheduledTime,
+					"-m", "auto",
+					"-y",
+				)
+				Expect(err).To(BeNil())
+				Expect(output.String()).To(ContainSubstring("Upgrade successfully scheduled for cluster"))
+			}
+
+			By("Check upgrade state")
+			err = WaitForUpgradeToState(upgradeService, clusterID, con.Scheduled, 4)
+			Expect(err).To(BeNil())
+			err = WaitForUpgradeToState(upgradeService, clusterID, con.Started, 70)
```

Comment:
Do we not wait for the cluster upgrading to be finished?

Reply:
According to our discussion, we verify the upgrade state currently and will enhance this part in the future.